lib/node_modules/@stdlib/array/base/cuevery-by-right/lib/assign.js

```diff
@@ -0,0 +1,172 @@
+	var flg;
+	var io;
+	var i;
+
+	flg = true;
+	io = offset;
+	for ( i = x.length - 1; i >= 0; i-- ) {
+		if ( flg === true && !predicate.call( thisArg, x[ i ], i, x ) ) {
+			flg = false;
+		}
+		out[ io ] = flg;
+		io += stride;
+	}
+	return out;
+}
+
+/**
+* Cumulatively test whether every array element in a provided accessor array pass a test implemented by a predicate function, while iterating from right-to-left and assign the results to elements in the accessor output array.
+*
+* @private
+* @param {Object} x - input array object
+* @param {Object} out - output array object
+* @param {integer} stride - output array stride
+* @param {NonNegativeInteger} offset - output array offset
+* @param {Function} predicate - test function
+* @param {*} [thisArg] - execution context
+* @returns {Collection} output array
+*
+* @example
+* function fcn( value ) {
+* 	 return ( value > 0 );
+* }
+*
+* var toAccessorArray = require( '@stdlib/array/base/to-accessor-array' );
+* var arraylike2object = require( '@stdlib/array/base/arraylike2object' );
+*
+* var x = toAccessorArray( [ 0, 0, 1, 1, 1 ] );
+*
+* var out = toAccessorArray( [ 0, 0, 0, 0, 0 ] );
+* var arr = accessors( arraylike2object( x ), arraylike2object( out ), 1, 0, fcn );
+*
+* var v = arr.get( 4 );
+* // returns false
+*/
+function accessors( x, out, stride, offset, predicate, thisArg ) {
+	var xdata;
+	var odata;
+	var xget;
+	var oset;
+	var flg;
+	var io;
+	var i;
+
+	xdata = x.data;
+	odata = out.data;
+
+	xget = x.accessors[ 0 ];
+	oset = out.accessors[ 1 ];
+
+	io = offset;
+	flg = true;
+	for ( i = xdata.length - 1; i >= 0; i-- ) {
+		if ( flg === true &&
+			!predicate.call( thisArg, xget( xdata, i ), i, xdata ) ) {
+			flg = false;
+		}
+		oset( odata, io, flg );
+		io += stride;
+	}
+	return odata;
+}
+
+
+// MAIN //
+
+/**
+* Cumulatively test whether every element in a provided array pass a test implemented by a predicate function, while iterating from right-to-left and assign the results to elements in the output array.
+*
+* @param {Collection} x - input array
+* @param {Collection} out - output array
+* @param {integer} stride - output array stride
+* @param {NonNegativeInteger} offset - output array offset
+* @param {Function} predicate - test function
+* @param {*} [thisArg] - execution context
+* @returns {Collection} output array
+*
+* @example
+* function fcn( value ) {
+* 	 return ( value > 0 );
+* }
+*
+* var x = [ 0, 0, 1, 1 ];
+*
+* var y = [ false, null, false, null, false, null, false, null ];
+* var out = assign( x, y, 2, 0, fcn );
+* // returns [ true, null, true, null, false, null, false, null ]
+*
+* var bool = ( y === out );
+* // returns true
+*/
+function assign( x, out, stride, offset, predicate, thisArg ) {
+	var xo;
+	var oo;
+
+	xo = arraylike2object( x );
+	oo = arraylike2object( out );
+	if (
+		xo.accessorProtocol ||
+		oo.accessorProtocol
+	) {
+		accessors( xo, oo, stride, offset, predicate, thisArg );
```

Comment:
It would be great to add support for Boolean and complex arrays here.

See e.g. https://github.com/stdlib-js/stdlib/blob/develop/lib/node_modules/%40stdlib/array/base/cuany/lib/assign.js for reference.